packages/cli-kit/src/log.ts

```diff
@@ -17,18 +17,42 @@ import {EOL} from 'node:os'
 
 const logFileName = 'shopify.cli.log'
 const maxLogFileSize = 5 * 1024 * 1024
+const maxLogFileSizeToTruncate = 30 * 1024 * 1024
 let logFileStream: WriteStream
 let commandUuid: string
 let logFilePath: string
 export class LinesTruncatorTransformer extends Transform {
   linesToRetain: string[] = []
   lastLineCompleted = true
-
-  constructor(readonly maxFileSize: number, opts?: TransformOptions) {
+  contentSize = 0
+
+  constructor(
+    readonly fileSize: number,
```

Comment:
Given there are several arguments, might be better to make these named parameters.